packages/core/useEventSource/index.ts

```diff
@@ -50,28 +50,28 @@ export interface UseEventSourceOptions extends EventSourceInit {
   autoConnect?: boolean
 }
 
-export interface UseEventSourceReturn<Events extends string[]> {
+export interface UseEventSourceReturn<Events extends string[], Data = any> {
   /**
    * Reference to the latest data received via the EventSource,
    * can be watched to respond to incoming messages
    */
-  data: Ref<string | null>
+  readonly data: Readonly<ShallowRef<Data>>
```

Comment:
~~Actually, it looks like there is a difference between `readonly data: ...` and `data: Readonly<...>'. 

https://www.typescriptlang.org/play/?#code/C4TwDgpgBAglC8UDeAoKUBOECGATA9gHYA2IUARgFzJrpQDG1AzsBgJaEDmtAvinylCQoAIQQ10VKACUcBEiAA8qOg2asO3dDwB8-FCgBmAV0L1gbIlGAQWMABTZqMAJQSo2AHTlP9cQCIbFn8ofSNTc0tCa1tgEXspETcVCm9fAKDgEP4gA

The return type of `shallowReadonly` is `Readonly<T>`, not sure if this matters.~~

looks like you already did. but do we need still the readonly prefix?

Reply:
there is a difference. it is why I wrote both

Reply:
`readonly` - you can't re-assign the property in the returned object
`Readonly` - you can't mutate the shallow ref

Reply:
so we always want to use both? :)

Reply:
pretty much and afaict, `shallowReadonly` has special treatment for if you give it a ref, so it makes the ref read-only and throws a warning if you change it

and `readonly` will return a `Readonly<Ref<T>>` rather than a `Readonly<ShallowRef<T>>` (doesn't matter much but means we lose the shallow type)